src/libs/Browser/index.web.js

```diff
@@ -50,8 +51,42 @@ function isMobileSafari() {
     return /iP(ad|od|hone)/i.test(userAgent) && /WebKit/i.test(userAgent) && !(/(CriOS|FxiOS|OPiOS|mercury)/i.test(userAgent));
 }
 
+function openRouteInDesktopApp(shortLivedAuthToken = '', email = '') {
+    const params = new URLSearchParams();
+    params.set('exitTo', `${window.location.pathname}${window.location.search}${window.location.hash}`);
+    if (email && shortLivedAuthToken) {
+        params.set('email', email);
+        params.set('shortLivedAuthToken', shortLivedAuthToken);
+    }
+    const expensifyUrl = new URL(CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL);
+    const expensifyDeeplinkUrl = `${CONST.DEEPLINK_BASE_URL}${expensifyUrl.host}/transition?${params.toString()}`;
+
+    // This check is necessary for Safari, otherwise, if the user
+    // does NOT have the Expensify desktop app installed, it's gonna
+    // show an error in the page saying that the address is invalid
```

Comment:
OK, interesting. Like, what lead you to try an iframe in the first place?

Reply:
It seems like someone had a similar problem here and there was a slightly different way of solving it. Could we try it out? https://stackoverflow.com/questions/18986200/suppress-safari-cant-open-the-page-because-the-address-is-invalid-custom-app-l

Also, do you know if the link URL matches this scheme? https://github.com/Expensify/App/blob/0b35d7c010157841617867ac3924699168cd9d27/ios/NewExpensify/Info.plist#L28

It looks like there are a few more tips in https://csiandal.medium.com/safari-cannot-open-the-page-because-the-address-is-invalid-2147d8963f89